cmd/demo/demo_test.go

```diff
@@ -46,40 +52,102 @@ func nodeCmd(name string) (*Cmd, error) {
 
 const blockTime = 5 * time.Second
 
+const sendIterations = 25
+
+const ethUrl = "ws://127.0.0.1:8545"
+
+const addressAlice = "0x05e71027e7d3bd6261de7634cf50F0e2142067C4"
+
+const addressBob = "0xA298Fc05bccff341f340a11FffA30567a00e651f"
+
 func TestNodes(t *testing.T) {
-	// Start Alice
+	// Start Alice.
 	alice, err := nodeCmd("alice")
 	require.NoError(t, err)
 	require.NoError(t, alice.Start())
 	defer alice.Process.Kill()
-	time.Sleep(blockTime * time.Duration(2)) // wait 2 blocks for contract deployment
+	time.Sleep(blockTime * 2) // Wait 2 blocks for contract deployment.
 
-	// Start Bob
+	// Start Bob.
 	bob, err := nodeCmd("bob")
 	require.NoError(t, err)
 	require.NoError(t, bob.Start())
 	defer bob.Process.Kill()
-	time.Sleep(5 * time.Second) // give bob some time to initialize
+	time.Sleep(5 * time.Second) // Give Bob some time to initialize.
+
+	// Set the initial on-chain balances from Alice and Bob.
+	initBals, err := getOnChainBals()
+	require.NoError(t, err)
+	t.Logf("Initial on-chain balances: Alice = %f, Bob = %f", initBals[0], initBals[1])
 
-	// Alice opens channel with Bob
+	// Alice opens channel with Bob.
 	require.NoError(t, alice.sendCommand("open bob 1000 1000\n"), "proposing channel")
+	time.Sleep(3 * time.Second) // Ensure that Bob really received the proposal.
 	require.NoError(t, bob.sendCommand("y\n"), "accepting channel proposal")
 	t.Log("Opening channel…")
-	time.Sleep(blockTime) // wait 1 block for funding transactions to be confirmed
+	time.Sleep(blockTime) // Wait 1 block for funding transactions to be confirmed.
 
-	// Alice sends to Bob and Bob to Alice
-	for i := 0; i < 25; i++ {
+	// Alice sends to Bob and Bob to Alice.
+	for i := 0; i < sendIterations; i++ {
 		t.Log("Sending payment… (alice->bob)")
 		require.NoError(t, alice.sendCommand("send bob 1\n"))
 		time.Sleep(100 * time.Millisecond)
 		t.Log("Sending payment… (bob->alice)")
 		require.NoError(t, bob.sendCommand("send alice 2\n"))
 		time.Sleep(100 * time.Millisecond)
 	}
+
+	t.Log("Closing channel…")
+	require.NoError(t, alice.sendCommand("close bob\n"))
+	// Wait 2 blocks for the settle and withdrawal transactions plus some additional seconds.
+	time.Sleep(2*blockTime + 5*time.Second)
+
+	// Set the final balances from Alice and Bob after the settlement.
+	finalBals, err := getOnChainBals()
+	require.NoError(t, err)
+
+	t.Logf("Final on-chain balances: Alice = %f, Bob = %f", finalBals[0], finalBals[1])
+
+	var diffBals [2]float64
+
+	// Calculate the differences between the initial and final balances.
+	diffBals[0], _ = finalBals[0].Sub(finalBals[0], initBals[0]).Float64()
+	diffBals[1], _ = finalBals[1].Sub(finalBals[1], initBals[1]).Float64()
+
+	// Check the on-chain balance differences while allowing a higher deviation for Alice.
+	require.Less(t, diffBals[0], sendIterations+0.1)
```

Comment:
You can use `assert.InEpsilon(t, sendIterations, diffBals[0], 0.1)` here and below, then you need just one check.
This is the only place where it is actually an assert instead of a require so it is probably not obvious :see_no_evil: 

I also tried to set the gas price of the chain to `0`, which then produces the exact values of `+-25.0` but i think it is better to run it with gas like you do right now.